core/src/main/scala/com/avast/clients/rabbitmq/DefaultRabbitMQClientFactory.scala

```diff
@@ -464,16 +467,17 @@ private[rabbitmq] object DefaultRabbitMQClientFactory extends LazyLogging {
     consumer
   }
 
-  private def wrapReadAction[A](consumerConfig: ConsumerConfig, userReadAction: DefaultDeliveryReadAction, blockingScheduler: Scheduler)(
-      implicit callbackScheduler: Scheduler): DefaultDeliveryReadAction = {
+  private def wrapReadAction[F[_]: Effect, A](consumerConfig: ConsumerConfig,
+                                              userReadAction: DefaultDeliveryReadAction[F],
+                                              blockingScheduler: Scheduler)(implicit sch: Scheduler): DefaultDeliveryReadAction[F] = {
     import consumerConfig._
 
     delivery: Delivery[Bytes] =>
       try {
         // we try to catch also long-lasting synchronous work on the thread
         val action = Task.deferFuture {
           Future {
-            userReadAction(delivery)
+            Task.fromEffect { userReadAction(delivery) }
```

Comment:
Are you sure this is correct? `Task[Future[Task]]]`?

Reply:
It is but I agree it looks weird (BTW it's actually just `Task[Task]`). I changed it (removed the `Future`).